src/rp2_common/hardware_dma/include/hardware/dma.h

```diff
@@ -556,7 +588,20 @@ static inline bool dma_channel_get_irq0_status(uint channel) {
  */
 static inline bool dma_channel_get_irq1_status(uint channel) {
     check_dma_channel_param(channel);
-    return dma_hw->ints0 & (1u << channel);
+    return dma_hw->ints1 & (1u << channel);
+}
+
+/*! \brief  Determine if a particular channel is a cause of DMA_IRQ_N
+ *  \ingroup hardware_dma
+ *
+ * \param irq_index the IRQ index; either 0 or 1 for DMA_IRQ_0 or DMA_IRQ_1
+ * \param channel DMA channel
+ * \return true if the channel is a cause of the DMA_IRQ_N, false otherwise
+ */
+static inline bool dma_irqn_get_channel_status(uint irq_index, uint channel) {
```

Comment:
Apologies if this is a dumb question, but what's the "logic" behind the naming of these dma functions?
```
dma_channel_set_irq0_enabled(uint channel, bool enabled)
dma_channel_set_irq1_enabled(uint channel, bool enabled)
dma_irqn_set_channel_enabled(uint irq_index, uint channel, bool enabled)

dma_set_irq0_channel_mask_enabled(uint32_t channel_mask, bool enabled)
dma_set_irq1_channel_mask_enabled(uint32_t channel_mask, bool enabled)
dma_set_irqn_channel_mask_enabled(uint irq_index, uint32_t channel_mask,  bool enabled)

dma_channel_get_irq0_status(uint channel)
dma_channel_get_irq1_status(uint channel)
dma_irqn_get_channel_status(uint irq_index, uint channel)

dma_channel_acknowledge_irq0(uint channel)
dma_channel_acknowledge_irq1(uint channel)
dma_irqn_acknowledge_channel(uint irq_index, uint channel)
```

Reply:
dma_irqX is preferred because the IRQ number is the target; the `dma_channel_s/get_irqX` were pre-existing, so I didn't rename, and I thought adding more aliases would be even more confusing. These pre-existing deal with only one IRQ and on e channel, so the difference isn't so great.

Reply:
there is a section on naming in the C++ book which is what I try to abide by, however there were some older ones that don't follow

Reply:
"dma_irqX is preferred... adding more aliases would be even more confusing."

Thanks, that's as good a rule as any. Does that mean `dma_set_irqn_channel_mask_enabled` ought to be `dma_irqn_set_channel_mask_enabled` ?

Reply:
yes!

Reply:
fixed